src/effects/backends/effectprocessor.h

```diff
@@ -148,6 +155,19 @@ class EffectProcessorImpl : public EffectProcessor {
         return 0;
     }
 
+    std::unique_ptr<DlgEffect> createUI() override {
```

Comment:
This seems wrong, as you declare this as pure virtual in this same file line 119: `virtual std::unique_ptr<DlgEffect> createUI() = 0;`

I would expect the GUI button disabled instead of showing a dialog that this effect has no GUI.

Reply:
> This seems wrong, as you declare this as pure virtual in this same file line

The pure virtual signature applies to `EffectProcessor`, which serves as the interface, this method is in `EffectProcessorImpl`

> I would expect the GUI button disabled instead of showing a dialog that this effect has no GUI.

We do that too, but there are other advantages to keeping the dialog available, see this discussion for rationale: https://github.com/mixxxdj/mixxx/pull/13888#discussion_r1843617674